src/net/rpc.rs

```diff
@@ -632,6 +632,12 @@ impl ConversationHttp {
                     response_metadata, "{ \"message\": \"Transaction not accepted.\"".into());
                 response.send(&mut self.connection.protocol, &mut reply).map(|_| ())?;
                 None
+            },
+            HttpRequestType::OptionsPreflight(ref _md) => {
+                let response_metadata = HttpResponseMetadata::from(&req);
+                let response = HttpResponseType::OptionsPreflight(response_metadata);
+                response.send(&mut self.connection.protocol, &mut reply).map(|_| ())?;
+                None
```

Comment:
Please add unit tests for the following:

* Parse and serialize Options requests
* Parse and serialize Options responses
* Check that Options responses contain the requisite CORS headers

Reply:
Also, looks like this code caused a regression in `test_http_response_type_codec`.  It's pretty easy to fix -- if we're including CORS headers on each response, then the test code at line 3358 should be expanded to test for the presence of each CORS header, in addition to `date` and `server` (and the quantity of expected headers should be updated).